framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -529,32 +537,142 @@ public TypeArgumentInference getTypeArgumentInference() {
     }
 
     /**
-     * If the type factory or checker class is annotated with {@link
-     * TypeQualifiers}, return an immutable set with the same set
-     * of classes as the annotation.  If the class is not so annotated,
-     * return an empty set.
+     * Returns an immutable set of annotation classes that are supported by a checker
```

Comment:
I agree to avoid duplication in general.  A comment in the Javadoc seems more appropriate to me in general if it is documenting what a particular method does.  It will be more available to programmers.  A mention, with a single sentence of description, in the manual seems appropriate.